firecrown/ccl/systematics/pz.py

```diff
@@ -37,3 +39,54 @@ def apply(self, cosmo, params, source):
                 source.z_ - params[self.delta_z], extrapolate=False)
         _dndz[np.isnan(_dndz)] = 0.0
         source.dndz_ = _dndz
+
+
+class PhotoZSystematic(Systematic):
+    """ A photo-z systematic.
+
+    Convolves the original redshift distribution with a gaussian filter
+    with mean mu and std sigma that evolve with redshift according to
+    mu_shift = z + mu_0 + (1+z)*mu_1 and sigma_shift = (1+z)*sigma
+
+    TODO: use more efficient integration and test more complex parametrizations
+
+    Parameters:
+    -----------
+    mu_0, mu_1: parameters that describes the evolution of the mean function
+    sigma: parameter that describes the evolution of the std function
+
+    Methods
+    -------
+    apply : apply the systematic to a source
+    """
+    def __init__(self, mu_0, mu_1, sigma):
+        self.mu_0 = mu_0
+        self.mu_1 = mu_1
+        self.sigma = sigma
+
+    def apply(self, cosmo, params, source):
+        """Apply the systematic to the photo-z distribution of a source.
+
+        Parameters
+        ----------
+        cosmo : pyccl.Cosmology
+            A pyccl.Cosmology object.
+        params : dict
+            A dictionary mapping parameter names to their current values.
+        source : a source object
+            The source to which apply the shift.
+        """
+        zorig, N = (source.z_, len(source.z_))
+        zpdf_orig = source.dndz_/np.sum(source.dndz_*source.z_)
```

Comment:
This confuses me. z_ is the array of z values, not the bin spacing. Also, dndz is already normalized here.